source/features/useful-not-found-page.tsx

```diff
@@ -36,46 +38,156 @@ function parseCurrentURL(): string[] {
 	return parts;
 }
 
-// If the resource was deleted, link to the commit history
-async function addCommitHistoryLink(bar: Element): Promise<void> {
+async function findAndDisplayMissingPart(bar: Element): Promise<void> {
+	// The branch might have been deleted
+	const branchDeleted = await addBranchStatus(bar);
+	// Or the file exists, but on a different branch
+	const branchDoesNotHaveFile = await addDefaultBranchLink(bar);
+	// If either of the above are true, we don't need to go into the file's commit history.
+	if (branchDeleted || branchDoesNotHaveFile) {
+		return;
+	}
+
+	// The object might have been deleted/moved
+	if (await addObjectStatus(bar)) {
+		return;
+	}
+
+	// Or 410 Gone
+	await addCommitHistoryLink(bar);
+}
+
+async function addObjectStatus(bar: Element): Promise<boolean> {
+	// Get the file path from the parts
+	const parts = parseCurrentURL();
+	const filePath = parts.slice(4).join('/');
+
+	// Get the last 2 commits that include the file
+	const previousCommitsIncludingFile = await api.v3(`commits?path=${filePath}&per_page=2`);
+	// The latest commit will be the first object in the array
+	if (previousCommitsIncludingFile[0]) {
+		// Get a list of changes that happened in the repo with this commit
+		const lastCommitInfo = await api.v3(`commits/${previousCommitsIncludingFile[0].sha as string}`);
```

Comment:
I forgot to mention we should use v4. We only use v3 if there's no alternative, which is once or twice.

Reply:
Sure, I'll look up v4 and change all the calls to use that.

Reply:
I changed the call on line 65 (retrieving commits for the file that wasn't found) to use the v4 API.

On line 70 though (now line 87), it seems there is [no way to get a list of files from the GraphQL API](https://github.community/t/graphql-api-get-list-of-files-related-to-commit/14047/2). I looked up the [API reference](https://docs.github.com/en/graphql/reference/objects#commit) too, and the `history` and `blame` properties on the `Commit` object do not have any fields that will allow us to access the status of each file. I have kept it to use v3 for now. 

Please let me know if I missed something, and it is possible to get the status of each file in a commit using v4.

Thanks!

Reply:
Indeed: https://github.com/sindresorhus/refined-github/blob/9b8ac9771b55c0e5f3d6ba53a4da2a3b77556717/source/features/follow-file-renames.tsx#L17